iwyu_ast_util.cc

```diff
@@ -1422,6 +1432,31 @@ FunctionDecl* GetRedeclSpecifyingDefArg(unsigned i, FunctionDecl* func) {
   CHECK_UNREACHABLE_("There should be a redecl specifying the default arg");
 }
 
+const TemplateDecl* GetRedeclSpecifyingDefArg(unsigned i,
+                                              const TemplateDecl* decl) {
+  const auto* redeclarable = dyn_cast<RedeclarableTemplateDecl>(decl);
+  // If this template isn't redeclarable, then the given declaration should
+  // specify default arguments, if any.
+  if (!redeclarable)
+    return decl;
+  for (const RedeclarableTemplateDecl* redecl : redeclarable->redecls()) {
+    const TemplateParameterList* params = redecl->getTemplateParameters();
+    if (IsDefTplArgSpecified(params->getParam(i)))
+      return redecl;
+  }
+  CHECK_UNREACHABLE_("There should be a redecl specifying the default arg");
```

Comment:
Could you add a little comment above this check to describe what I did wrong if it fails? I'm guessing this function can only be called with `TemplateDecl`s for which we know there are default parameters, or something?

Reply:
I see, I hadn't gotten that far in the diff when I read this.